src/main/scala/is/hail/check/Gen.scala

```diff
@@ -404,8 +406,12 @@ class Gen[+T](val gen: (Parameters) => T) extends AnyVal {
   // FIXME should be non-strict
   def withFilter(f: (T) => Boolean): Gen[T] = Gen { (p: Parameters) =>
     var x = apply(p)
-    while (!f(x))
+    var i = 0
+    while (!f(x)) {
+      assert(i < 100)
```

Comment:
If you filter too heavily, the tests will be slow (or, in the limit, go into an infinite loop).